src/Symfony/Component/Serializer/Dumper/NormalizerDumper.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Dumper;
+
+use Symfony\Component\Serializer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Mapping\Factory\ClassMetadataFactoryInterface;
+
+/**
+ * @author Guilhem Niot <guilhem.niot@gmail.com>
+ */
+final class NormalizerDumper
+{
+    private $classMetadataFactory;
+
+    public function __construct(ClassMetadataFactoryInterface $classMetadataFactory)
+    {
+        $this->classMetadataFactory = $classMetadataFactory;
+    }
+
+    public function dump($class, array $context = array())
+    {
+        $reflectionClass = new \ReflectionClass($class);
+        if (!isset($context['class'])) {
+            $context['class'] = $reflectionClass->getShortName().'Normalizer';
+        }
+
+        $namespaceLine = isset($context['namespace']) ? "\nnamespace {$context['namespace']};\n" : '';
+
+        return <<<EOL
+<?php
+$namespaceLine
+use Symfony\Component\Serializer\Exception\CircularReferenceException;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+
+/**
+ * This class is generated.
+ * Please do not update it manually.
+ */
+class {$context['class']} implements NormalizerInterface, NormalizerAwareInterface
+{
+    use NormalizerAwareTrait;
+
+{$this->generateNormalizeMethod($reflectionClass)}
+
+{$this->generateSupportsNormalizationMethod($reflectionClass)}
+}
+EOL;
+    }
+
+    /**
+     * Generates the {@see NormalizerInterface::normalize} method.
+     *
+     * @param \ReflectionClass $reflectionClass
+     *
+     * @return string
+     */
+    private function generateNormalizeMethod(\ReflectionClass $reflectionClass)
+    {
+        return <<<EOL
+    public function normalize(\$object, \$format = null, array \$context = array())
+    {
+{$this->generateNormalizeMethodInner($reflectionClass)}
+    }
+EOL;
+    }
+
+    private function generateNormalizeMethodInner(\ReflectionClass $reflectionClass)
+    {
+        $code = <<<EOL
+
+        \$objectHash = spl_object_hash(\$object);
+        if (isset(\$context[ObjectNormalizer::CIRCULAR_REFERENCE_LIMIT][\$objectHash])) {
+            throw new CircularReferenceException('A circular reference has been detected (configured limit: 1).');
```

Comment:
About this, I've experimented better use for it by using `return null` instead of rendering the normalization process impossible while using circular data.